packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -1508,35 +1538,45 @@ export class WindowsBuildImage implements IBuildImage {
    * @param repository The ECR repository
    * @param tag Image tag (default "latest")
    */
-  public static fromEcrRepository(repository: ecr.IRepository, tag: string = 'latest'): IBuildImage {
+  public static fromEcrRepository(
+    repository: ecr.IRepository,
+    tag: string = 'latest',
+    imageType: WindowsImageType = WindowsImageType.STANDARD): IBuildImage {
     return new WindowsBuildImage({
       imageId: repository.repositoryUriForTag(tag),
       imagePullPrincipalType: ImagePullPrincipalType.SERVICE_ROLE,
+      imageType,
       repository,
     });
   }
 
   /**
    * Uses an Docker image asset as a Windows build image.
    */
-  public static fromAsset(scope: Construct, id: string, props: DockerImageAssetProps): IBuildImage {
+  public static fromAsset(
+    scope: Construct,
+    id: string,
+    props: DockerImageAssetProps,
+    imageType: WindowsImageType = WindowsImageType.STANDARD): IBuildImage {
     const asset = new DockerImageAsset(scope, id, props);
     return new WindowsBuildImage({
       imageId: asset.imageUri,
       imagePullPrincipalType: ImagePullPrincipalType.SERVICE_ROLE,
+      imageType,
       repository: asset.repository,
     });
   }
 
-  public readonly type = 'WINDOWS_CONTAINER';
   public readonly defaultComputeType = ComputeType.MEDIUM;
+  public readonly type: string;
   public readonly imageId: string;
   public readonly imagePullPrincipalType?: ImagePullPrincipalType;
   public readonly secretsManagerCredentials?: secretsmanager.ISecret;
   public readonly repository?: ecr.IRepository;
 
   private constructor(props: WindowsBuildImageProps) {
     this.imageId = props.imageId;
+    this.type = props.imageType?.toString() ?? WindowsImageType.STANDARD.toString();
```

Comment:
You can save one `.toString()` by going

```suggestion
    this.type = (props.imageType ?? WindowsImageType.STANDARD).toString();
```